### PR TITLE
SETUP.md and TESTING.md updates for first-time success

### PR DESCRIPTION
**Pull request by snickell**
Patchups to `SETUP.md` and `TESTING.md` to make the instructions work for first-timers, particularly on macOS Monterey.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -128,13 +130,16 @@ These steps may need to change over time as 3rd party tools update to have versi
           4. Confirm MySQL has started by running `brew services` again.
 
 1. Install the **Java 8 JSK**
-   1. Either explicitly via `brew cask install adoptopenjdk/openjdk/adoptopenjdk8` or for M1 in Rosetta, `brew install --cask adoptopenjdk/openjdk/adoptopenjdk8`
+   1. Either explicitly via `brew install --cask adoptopenjdk/openjdk/adoptopenjdk8` or for M1 in Rosetta, `brew install --cask adoptopenjdk/openjdk/adoptopenjdk8`
```
Comment (mgc1194) [inline SETUP.md]: The command looks the same in both cases, does it need to be duplicated?

File: SETUP.md [hunk 2]
```diff
@@ -25,6 +25,8 @@ You can do Code.org development using OSX, Ubuntu, or Windows (running Ubuntu in
 
 1. `rbenv rehash`
 
+1. `bundle config --local without staging test production levelbuilder`
```
Comment (davidsbailey) [inline SETUP.md:30]: one alternative to consider would be to just run `bundle install --without staging test production levelbuilder` on the next line and remove this line. the `rake install` step a few lines later should set the appropriate bundle config:
* https://github.com/code-dot-org/code-dot-org/blob/2203800f49d92f55d18501a3e79fe4e3f2ccfc3f/lib/rake/install.rake#L46
https://github.com/code-dot-org/code-dot-org/blob/14946a976e9f27d51ff0cc8016810feefa28efe5/lib/cdo/rake_utils.rb#L154-L156

Comment (snickell) [reply SETUP.md:30]: I spent a little time before suggesting this direction exploring whether there was a clean trick for running the rake task BEFORE the first bundle install, but as @elijah guessed initially its a chicken and egg problem as currently configured

File: SETUP.md (hunk 1 above)
Comment (snickell) [reply SETUP.md]: Great point! I didn't notice that, a number of "for M1" sections are actually not M1 specific but instead specific to newer versions of macOS.

**Review by davidsbailey — APPROVED** [review]
Thanks for making this improvement, Seth! The scenario of installing from scratch definitely slipped through the cracks when we looked at this last.

File: SETUP.md (hunk 2 above)
Comment (davidsbailey) [reply SETUP.md:30]: I would still expect `bundle install --without staging test production levelbuilder` to succeed before running `rake install` or `bundle config ...`, but I could be mistaken. if you've already proved this way works then I'm fine with merging as is.

Comment (snickell) [reply SETUP.md:30]: `bundle install --without` works, but is listed as deprecated. I preferred that too, but I figure no point putting in a deprecated command even if its cleaner for now.

Comment (davidsbailey) [reply SETUP.md:30]: ahh, I missed that. Nice catch! looks good as-is then :-)

**Comment by snickell** [comment]
TY for help getting my build going :tada:

I’ve moved on to `TESTING.md`. It appears that the dashboard tests require mini_racer, so if a working dev setup for dashboard entails running tests (probably? lmk if not), working around installing mini_racer on mac may not get us/me all the way.

Long-term:
@davidsbailey has a jira issue open for the “install node on prod” long-term fix: see https://codedotorg.atlassian.net/browse/INF-708 and https://github.com/code-dot-org/code-dot-org/pull/47788

Short-term:
Perhaps the easiest workaround for SETUP.md, pending resolution of the jira issue, would be to suggest mac devs maintain an unstaged change bumping the `mini_racer` version in `Gemfile.lock` to 0.5.0? Its ugly, but its probably (?) better to acknowledge ugly and have setup instructions that get new devs all the way.

Thoughts?

**Review by mgc1194 — APPROVED** [review]
Looks Great, less confusing and easier to follow!

**Pull request merged**